plugins/elasticsearch-bboss/src/main/java/com/navercorp/pinpoint/plugin/elasticsearchbboss/ElasticsearchPlugin.java

```diff
@@ -0,0 +1,202 @@
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		interceptorClassInfo.setAllAccept(methodInfo);
+
+		methodInfos = null;
+		methodInfos = new ArrayList<ElasticsearchMethodInfo>();
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		methodInfos.add(methodInfo);
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(0);
+		methodInfo.setName("discover");
+		methodInfo.setPattern(false);
+		methodInfos.add(methodInfo);
+		interceptorClassInfo.setInterceptorMehtods(methodInfos);
+		interceptorClassInfo.setMethodFilter(new ElasticsearchCustomMethodFilter(null, interceptorClassInfo));
+		clazzInterceptors.add(interceptorClassInfo);
+		interceptorClassInfo = new ElasticsearchInterceptorClassInfo();
+		interceptorClassInfo.setInterceptorClass("org.frameworkset.elasticsearch.client.RestClientUtil");
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		interceptorClassInfo.setAllAccept(methodInfo);
+
+		methodInfos = null;
+		methodInfos = new ArrayList<ElasticsearchMethodInfo>();
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(1);
+		methodInfo.setName("*");
+		methodInfo.setPattern(true);
+		methodInfos.add(methodInfo);
+		methodInfo = new ElasticsearchMethodInfo();
+		methodInfo.setFilterType(0);
+		methodInfo.setName("discover");
+		methodInfo.setPattern(false);
+		methodInfos.add(methodInfo);
+		interceptorClassInfo.setInterceptorMehtods(methodInfos);
+		interceptorClassInfo.setMethodFilter(new ElasticsearchCustomMethodFilter(null, interceptorClassInfo));
+		clazzInterceptors.add(interceptorClassInfo);
+	}
+
+	public static List<ElasticsearchInterceptorClassInfo> getClazzInterceptors(){
+		return clazzInterceptors;
+	}
+
+	private final PLogger logger = PLoggerFactory.getLogger(this.getClass());
+	private TransformTemplate transformTemplate;
+	private RestSeachExecutorMethodFilter restSeachExecutorMethodFilter = new RestSeachExecutorMethodFilter();
+
+	@Override
+	public void setup(ProfilerPluginSetupContext context) {
+		if (context == null) {
+			return;
+		}
+
+		ElasticsearchPluginConfig elasticsearchPluginConfig = new ElasticsearchPluginConfig(context.getConfig());
+		if (logger.isInfoEnabled()) {
+			logger.info("ElasticsearchPlugin config:{}", elasticsearchPluginConfig);
+		}
+
+		if (!elasticsearchPluginConfig.isEnabled()) {
+			return;
+		}
+		this.addApplicationTypeDetector(context);
+		addElasticsearchInterceptors();
+		addElasticsearchExecutorInterceptors();
+	}
+
+	/**
+	 * Pinpoint profiler agent uses this detector to find out the service type of current application.
+	 */
+	private void addApplicationTypeDetector(ProfilerPluginSetupContext context) {
+		context.addApplicationTypeDetector(new ElasticsearchProviderDetector(clazzInterceptors));
+	}
+
```

Comment:
Will  _addApplicationTypeDetector_ be needed for elasticsearchbboss plugin?
If there is any reason that's on your mind let me know :)

Reply:
Elasticsearchbboss plugin maybe an server type plugin if there is not server container as a server endpoint. So I think that the AddApplicationTypeDetector is needed for elasticsearchbboss plugin.